### PR TITLE
Orthogonal C7

### PR DESCRIPTION
**Pull request by TristanPfersdorff**
See comment below for lack of test coverage. Tests for `gap/ClassicalMaximals.gi` are up next in #115.

### CODE REVIEW

**Comment by fingolfin** [comment]
When you say "too large to cover with tests" what do you mean? That you cannot even create the group because it is too slow? Or that running group recognition is too slow? In the former case, let's try to make it faster otherwise it is useless. In the latter case: simply don't try group recognition (for now), but we can still perform the other tests (e.g. is the form preserved, are the determinants right etc.)

File: gap/TensorInducedMatrixGroups.gi [hunk 1]
```diff
@@ -286,3 +286,278 @@ function(m, t, q)
...
+            # add it here.
+            I := IdentityMat(m ^ (t - 2), field);
+            D := NormSpMinusSp(m, q);
+            Add(gens, KroneckerProduct(KroneckerProduct(D, Inverse(D)), I));
+        fi;
+
+
+        F := AntidiagonalMat(Concatenation(ListWithIdenticalEntries(m / 2, one),
+                                        ListWithIdenticalEntries(m / 2, -one)),
+                                        field);
```
Comment (fingolfin) [inline gap/TensorInducedMatrixGroups.gi:364]: See PR #116

File: gap/ClassicalMaximals.gi [hunk 2]
```diff
@@ -1543,6 +1543,106 @@ function(epsilon, n, q)
...
+                    numberOfConjugates := 1;
+                elif t = 3 and m mod 4 = 2 and q mod 4 = 3 then
+                    numberOfConjugates := 2;
+                else
+                    numberOfConjugates := 4;
+                fi;
+
+                Append(result, ConjugateSubgroupOmega(1, n, q,
+                                                      OrthogonalEvenTensorInducedDecompositionStabilizerInOmega(1, m, t, q),
+                                                      numberOfConjugates));
```
[the remark this answers is not in the document]
Comment (TristanPfersdorff) [reply gap/ClassicalMaximals.gi]: Admittedly, readability in `gap/ClassicalMaximals.gi` has suffered somewhat, so I've decided to go through the orthogonal code and clean up a bit. It's still not perfect but a lot nicer on the eyes.

**Comment by TristanPfersdorff** [comment]
> When you say "too large to cover with tests" what do you mean? That you cannot even create the group because it is too slow? Or that running group recognition is too slow? In the former case, let's try to make it faster otherwise it is useless. In the latter case: simply don't try group recognition (for now), but we can still perform the other tests (e.g. is the form preserved, are the determinants right etc.)

I hadn't thought about this very thoroughly and I'm glad you brought it up, because I've run some tests concerning `OrthogonalEvenTensorInducedDecompositionStabilizerInOmega` and the results are very interesting. Test times are on my machine, a decent laptop:
- We can actually easily cover the else-branch with the input parameters `1, 8, 2, 3` in about 10sec, that was just an oversight on my part
- The main problem is the branch `elif t = 3 and m mod 4 = 2 and not squareDiscriminant`, and this is where things get interesting: the lightest input parameters for this case are `1, 6, 3, 3`, the testing splits into these steps:
	* constructing the group: about 100sec, everything here is pretty much instant apart from conjugation in the last line of the construction
	* checking for membership in Omega: about 90sec, pretty much all time is spent calculating the Witt Index of our form
	* checking size with RECOG: about 60sec
	


Fortunatly, the tests both pass.
I'm assuming we don't just want to throw in a 4-minute test in our CI, but there is one thing we can do: While calculating the Witt Index takes about 80-90sec here, the three functions/properties/attributes/filters (whichever term is correct here, even the forms package can't decide) `IsHyperbolicForm`, `IsParabolicForm` and `IsEllipticForm` from the forms package are near instant, so is `IsDegenerateForm` from the same package. I honestly have no idea why `WittIndex` is so much slower than all of these functions, but it is what it is.
In conclusion: One of the cases is really slow to test (about 4min on my machine), but if we change `CheckSubsetOmega` from `tst/utils.g`, we could speed it up to about 2.5min (still bad, but a little less bad) and also probably speed up other tests, albeit `CheckSubsetOmega` would look slightly uglier. Conjugation is also really slow, maybe we can do something about that too.

**Review by fingolfin — APPROVED** [review]
